postgresql/resource_postgresql_role.go

```diff
@@ -961,6 +1020,43 @@ func grantRoles(txn *sql.Tx, d *schema.ResourceData) error {
 	return nil
 }
 
+func setConfigurationParameters(txn *sql.Tx, d *schema.ResourceData) error {
+	role := d.Get(roleNameAttr).(string)
+	if d.HasChange(roleParameterAttr) {
+		o, n := d.GetChange(roleParameterAttr)
+		oldParams := o.(*schema.Set)
+		newParams := n.(*schema.Set)
+		for _, p := range oldParams.List() {
+			if !newParams.Contains(p) {
```

Comment:
Curiosity question: I'm assuming `Contains` has to deal with some sort of equality check. Is that just not something we have to worry about? (In terms of what defines "=" to understand if something's contained in a list or not)

Seems like just defining the schema of "parameter" + marking the role as containing a "set" of them was enough to let terraform understand what's equal and what's not. Is that fair to say?

Reply:
That's exactly what the [first argument](https://github.com/Leafly-com/terraform-provider-postgresql/pull/10/files#diff-ac561a01e0eaf9cc0ab03f251f4b154dc46b386eec3fa86b6ec72f7c13b76ad9R602) to `schema.NewSet()` does — it defines the hashing function used to populate the set. In this case, since the set contains sub-resources, we use `schema.HashResource()` to generate a hashing function using the resource's definition. (This is why the sub-resource had to be declared in a separate function, so it could be referenced there.)